provider/provider.go

```diff
@@ -2,38 +2,181 @@ package provider
 
 import (
 	"context"
-	"fmt"
+	"os"
 	"regexp"
 	"strings"
 
 	"github.com/cbroglie/mustache"
+	"github.com/getkin/kin-openapi/openapi3"
 	"github.com/go-logr/logr"
 	"github.com/hashicorp/go-version"
-	"github.com/konveyor/analyzer-lsp/dependency/dependency"
-	depprovider "github.com/konveyor/analyzer-lsp/dependency/provider"
 	"github.com/konveyor/analyzer-lsp/engine"
-	"github.com/konveyor/analyzer-lsp/provider/builtin"
-	"github.com/konveyor/analyzer-lsp/provider/golang"
-	"github.com/konveyor/analyzer-lsp/provider/java"
-	"github.com/konveyor/analyzer-lsp/provider/lib"
 	"github.com/konveyor/analyzer-lsp/tracing"
+	"go.lsp.dev/uri"
 	"go.opentelemetry.io/otel/attribute"
 	"gopkg.in/yaml.v2"
 )
 
+// This will need a better name, may we want to move it to top level
+// Will be used by providers for common interface way of passing in configuration values.
+var builtinConfig = Config{
+	Name: "builtin",
+}
+
+func init() {
+	c, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	builtinConfig.InitConfig = InitConfig{
+		Location: c,
+	}
+}
+
+type Capability struct {
+	Name            string
+	TemplateContext openapi3.SchemaRef
+}
+
+type Config struct {
+	Name       string     `yaml:"name,omitempty"`
+	BinaryPath string     `yaml:"binaryPath,omitempty"`
+	Address    string     `yaml:"address,omitempty"`
+	InitConfig InitConfig `yaml:"initConfig,omitempty"`
```

Comment:
TODO: Add Issue for multiple Init Configs

Reply:
created #158  

I think this is going to cause use some more changes, can we do this as a follow-up?

@pranavgaikwad @fabianvf